### PR TITLE
Added few more options to `DiskS3` client configuration

### PR DESCRIPTION
**Pull request by excitoon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Added few more options to `DiskS3` client configuration.

### CODE REVIEW

File: src/Disks/S3/registerDiskS3.cpp [hunk 1]
```diff
@@ -118,6 +118,9 @@ void registerDiskS3(DiskFactory & factory)
 
         cfg.connectTimeoutMs = config.getUInt(config_prefix + ".connect_timeout_ms", 10000);
         cfg.httpRequestTimeoutMs = config.getUInt(config_prefix + ".request_timeout_ms", 5000);
+        cfg.maxConnections = config.getUInt(config_prefix + ".max_connections", 25);
+        cfg.followRedirects = config.getBool(config_prefix + ".follow_redirects", false);
```
Comment (alesapin) [inline src/Disks/S3/registerDiskS3.cpp:122]: Why we set it to false? The default value is true https://github.com/ClickHouse-Extras/aws-sdk-cpp/blob/master/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp#L66

**Comment by Jokser** [comment]
Btw, all of these options are not used in the current HTTP client yet.
I think it makes sense to add real usage of them as well.

**Comment by excitoon** [comment]
> Btw, all of these options are not used in the current HTTP client yet.
> I think it makes sense to add real usage of them as well.

What is current HTTP client? :wow:

https://github.com/ClickHouse/ClickHouse/blob/master/contrib/aws-s3-cmake/CMakeLists.txt#L94

**Comment by Jokser** [comment]
@excitoon Current HTTP client is https://github.com/ClickHouse/ClickHouse/blob/master/src/IO/S3/PocoHTTPClient.cpp . The reason for implementing this on Poco was memory issue when all response is copied into memory

**Comment by alexey-milovidov** [comment]
@Jokser Maybe we can get rid of `curl` library in our build?

**Comment by Jokser** [comment]
@alexey-milovidov As I can see `curl` is used as a dependency in `mariadb-connector` and `sentry-native`

**Comment by alesapin** [comment]
Closed because this functionality implemented in other PR (I don't know which one).

**Pull request closed without merging**